tensorflow/python/eager/backprop.py

```diff
@@ -811,6 +827,9 @@ def watch(self, tensor):
     Args:
       tensor: a Tensor or list of Tensors.
     """
+    if check_float_dtype(tensor) == False:
+        logging.log_first_n(logging.WARN, "The expected Datatype must be 'tf.float' "
```

Comment:
Is there a specific reason to log just on the first occurrence? I think logging for every `gradient` call is OK.

Consider making the error message a bit more specific, something like

```python
"The expected dtype of the watched tensor must be floating (e.g. tf.float32), got {!r}".format(t.dtype)
```

Reply:
Yeah, just consider maybe too much warning there, just change it as you say. Thanks.